gcc/rust/backend/rust-compile-intrinsic.cc

```diff
@@ -0,0 +1,105 @@
+// This file is part of GCC.
+
+// GCC is free software; you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 3, or (at your option) any later
+// version.
+
+// GCC is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with GCC; see the file COPYING3.  If not see
+// <http://www.gnu.org/licenses/>.
+
+#include "rust-compile-intrinsic.h"
+
+namespace Rust {
+namespace Compile {
+
+Intrinsics::Intrinsics (Context *ctx) : ctx (ctx) {}
+
+Bfunction *
+Intrinsics::compile (TyTy::FnType *fntype)
+{
+  rust_assert (fntype->get_abi () == TyTy::FnType::ABI::INTRINSIC);
+
+  // https://github.com/rust-lang/rust/blob/master/compiler/rustc_codegen_llvm/src/intrinsic.rs
+  // https://github.com/Rust-GCC/gccrs/issues/658
+
+  std::map<std::string, std::string> mappings;
+  mappings["sqrtf32"] = "__builtin_sqrtf";
+  // mappings["sqrtf64"] = "__builtin_sqrtl";
+  mappings["sinf32"] = "__builtin_sinf";
```

Comment:
I think this is a nice way of doing this. Could it be refactored with...

Reply:
Yeah i think you are right I have changed it do do this now, which has also solved the need to link against -lm since it seemed to be falling back instead of using the buiiltin.